### PR TITLE
Adds support for cleaning up created certificates on the machine

### PR DESCRIPTION
**Pull request by javiercn**
This change adds the CleanupHttpsCertificates functionality on the core library that drives the UI. A follow up change will add the interface on the command-line tool.
  
Fixes https://github.com/aspnet/DotNetTools/issues/366

### CODE REVIEW

File: shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs [hunk 1]
```diff
@@ -33,9 +33,13 @@ internal class CertificateManager
         public const int RSAMinimumKeySizeInBits = 2048;
 
         private static readonly TimeSpan MaxRegexTimeout = TimeSpan.FromMinutes(1);
-        private const string MacOSFindCertificateCommandLine = "security";
-        private const string MacOSFindCertificateCommandLineArguments = "find-certificate -c localhost -a -Z -p /Library/Keychains/System.keychain";
+        private const string MacOSSystemKeyChain = "/Library/Keychains/System.keychain";
+        private static readonly string MacOSUserKeyChain = Environment.GetEnvironmentVariable("HOME") + "/Library/Keychains/login.keychain-db";
```
Comment (mkArtakMSFT) [inline shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs]: How will we ever (in the future) handle a situation where the key chain name or location changes in the future versions of MacOS?

Comment (javiercn) [reply shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs]: That will be a huge all breaking changes in OSX, but in any case, we would update the tool at that point.

File: shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs [hunk 2]
```diff
@@ -387,34 +402,181 @@ public bool IsTrusted(X509Certificate2 certificate)
...
+                    MacOSRemoveCertificateTrustCommandLine,
+                    string.Format(
+                        MacOSRemoveCertificateTrustCommandLineArgumentsFormat,
+                        certificatePath
+                    ))
+                {
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true
+                };
+                var process = Process.Start(processInfo);
```
Comment (mkArtakMSFT) [inline shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs]: process will be disposable, please wrap in a using statement

File: shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs [hunk 3]
```diff
@@ -387,34 +402,181 @@ public bool IsTrusted(X509Certificate2 certificate)
...
+                process.WaitForExit();
+                var output = process.StandardOutput.ReadToEnd() + process.StandardError.ReadToEnd();
```
Comment (mkArtakMSFT) [inline shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs]: You should read the output/error before WaitForExit() - as that can result in a deadlock: https://msdn.microsoft.com/en-us/library/system.diagnostics.process.standardoutput(v=vs.110).aspx

Comment (javiercn) [reply shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs]: Yeah, for some reason I thought it was the other way around.

File: shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs (hunk 1 above)
Comment (mkArtakMSFT) [reply shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs]: Ok

File: shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs [hunk 4]
```diff
@@ -378,43 +396,206 @@ public bool IsTrusted(X509Certificate2 certificate)
...
+                var certBytes = certificate.Export(X509ContentType.Cert);
+                File.WriteAllBytes(certificatePath, certBytes);
+                var processInfo = new ProcessStartInfo(
+                    MacOSRemoveCertificateTrustCommandLine,
+                    string.Format(
+                        MacOSRemoveCertificateTrustCommandLineArgumentsFormat,
+                        certificatePath
+                    ))
+                {
+                    RedirectStandardOutput = true,
```
Comment (mkArtakMSFT) [inline shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs]: Looks like you don't need to redirect output in this case

File: shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs [hunk 5]
```diff
@@ -378,43 +396,206 @@ public bool IsTrusted(X509Certificate2 certificate)
...
+                    break;
+                case RemoveLocations.All:
+                    if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                    {
+                        RemoveCertificateFromTrustedRoots(certificate);
+                    }
+                    RemoveCertificateFromUserStore(certificate);
+                    break;
+                default:
+                    throw new InvalidOperationException($"Invalid location.");
```
Comment (javiercn) [inline shared/Microsoft.AspNetCore.Certificates.Generation.Sources/CertificateManager.cs]: Unnecesary $

**Pull request merged**